src/Google/NotificationsService.php

```diff
@@ -34,12 +38,34 @@ class NotificationsService implements Service {
 	 */
 	private $route;
 
+	/**
+	 * The product repository dependency
+	 *
+	 * @var ProductRepository $product_repository
+	 */
+	protected $product_repository;
+
+	/**
+	 * The product helper dependency
+	 *
+	 * @var ProductHelper $product_helper
+	 */
+	protected $product_helper;
+
 	/**
 	 * Class constructor
+	 *
+	 * @param ProductRepository $product_repository
+	 * @param ProductHelper     $product_helper
 	 */
-	public function __construct() {
-		$blog_id     = Jetpack_Options::get_option( 'id' );
-		$this->route = "https://public-api.wordpress.com/wpcom/v2/sites/{$blog_id}/partners/google/notifications";
+	public function __construct(
+		ProductRepository $product_repository,
+		ProductHelper $product_helper,
```

Comment:
I'm wondering if these dependencies make sense. I've already suggested why we might not need a query in other comments.

But if the `notify` function is agnostic of what type of notification it is then shouldn't the calling class handle any specifics about the notification type? Setting the "after notification status" can easily be handled by the `Product NotificationJob` which knows that it's working with a product.

Otherwise eventually we are going to need to keep increasing the dependencies for this service to have product helpers, shipping rate helpers, etc...

Reply:
Adjusted here https://github.com/woocommerce/google-listings-and-ads/pull/2202/commits/89400673971356a0d34b2ca3d74f2b329e1b5ee8